cmd/manager/main.go

```diff
@@ -214,7 +217,12 @@ func newRootCommand() *cobra.Command {
 				log.Info("Starting the Cmd.")
 
 				// Start the Cmd
-				log.Fatal(mgr.Start(signals.SetupSignalHandler()))
+				err = mgr.Start(signals.SetupSignalHandler())
```

Comment:
We are missing the part of leader election where we ensure that the controllers are stopped before releasing the leader-election lock. The following order of events should occur.
1) TERM signal is received.
2) Object is sent to managers stop channel.
3) Managers stop.
4) Object is sent to leader-election stop channel.
5) Leader election lock is released.

Reply:
I may be wrong about this but looks like stop channel for manager returned by signals.SetupSignalHandler() is a receive only channel. So I think we cannot send anything on it to ensure controllers are stopped.

Reply:
Yes, you cannot use `signals.SetupSignalHandler`. You need to create your own channel to use instead.